components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/dashboard/HeartBeatComponent.java

```diff
@@ -123,7 +124,7 @@ public static void invokeHeartbeatExecutorService() {
                     ArtifactUpdateListener.removeFromUpdatedArtifactQueue(updatedArtifactsCount);
                 }
             } catch (IOException | NoSuchAlgorithmException | KeyStoreException | KeyManagementException e) {
```

Comment:
Hi, can I know why we are not catching all the exceptions at least as a fail safe? just asking because say next time some other unseen exception happens, this exact scenario will repeat, any particular reason for not catching everything and continue the thread?

Reply:
We did this to avoid unexpected exceptions. Catching _Exceptions_ usually not a good practice.

Reply:
I'm not an expert, but I would say, that depends on the scenario, like this one for example, we would need the thread to keep retrying whatever happens. Otherwise, the connection between the container and dashboard may disappear without any trace (nothing is even being logged)

Reply:
Agreed. This might create a issue in the future. I will create a Git issue to discuss this further.

Reply:
I opened https://github.com/wso2/api-manager/issues/1212 to fix this issue.